doc: config_and_build: add table for CMake options (#14518)
nrfconnect/sdk-nrf: doc/nrf/templates/sample_README.rst

   * Add this section to describe and link to any library configuration options that might be important to run this sample.
     You can link to options with ``:kconfig:option:`CONFIG_XXX```.
   * You do not need to list all possible configuration options, but only the ones that are relevant.

>    * If you need to provide specific steps with configuration variables, include them in a table and provide link to :ref:`cmake_options` for instructions on how to provide these build variables.

Reviewer: ```suggestion
   * If you need to provide specific steps with configuration variables, include them in a table and provide a link to :ref:`cmake_options` for instructions on how to provide these build variables.
```

     This is to avoid duplicating building instructions.
     The column entries for hardware platform and build target are optional.
     For example:

     .. list-table::
        :header-rows: 1

        * - Hardware platform*
          - Build target*
          - Files to use
          - Build variables
        * - nRF7002 DK
          - ``nrf7002dk_nrf5340_cpuapp``
          - :file:`overlay-nrf700x.conf`
          - ``-DEXTRA_CONF_FILE=overlay-nrf700x.conf``
        * - nRF7002 EK with nRF5340 DK
          - ``nrf5340dk_nrf5340_cpuapp``
          - :file:`overlay-nrf700x.conf`
          - | * ``-DEXTRA_CONF_FILE=overlay-nrf700x.conf``
            | * ``-DSHIELD=nrf7002ek``

>      Include "See :ref:`cmake_options` for instructions on how to provide these build variables" at the end of the table.

Reviewer: ```suggestion
     Include "See :ref:`cmake_options` for instructions on how to provide these build variables." as the first line after the table.
```
